pkg/nodeconfig/nodeconfig.go

```diff
@@ -288,6 +304,186 @@ func (nc *nodeConfig) Configure() error {
+	if err != nil {
+		return err
+	}
+	kubeconfigFileName := "bootstrap-kubeconfig"
+	if err := nc.Windows.EnsureFileContent(kubeconfigData, kubeconfigFileName, windows.K8sDir); err != nil {
+		return err
+	}
+	nc.log.V(1).Info("created", "file", windows.BootstrapKubeconfig)
+	return nil
+}
+
+// generateKubeconfig creates a kubeconfig spec with the certificate and token data from the given secret
+func generateKubeconfig(secret *core.Secret, apiServerURL string) (*clientcmdv1.Config, error) {
+	// extract ca.crt and token data fields
+	caCert := secret.Data[core.ServiceAccountRootCAKey]
+	if caCert == nil {
+		return nil, errors.New("unable to find CA cert")
+	}
+	token := secret.Data[core.ServiceAccountTokenKey]
+	if token == nil {
+		return nil, errors.New("unable to find token")
+	}
+	kubeconfig := clientcmdv1.Config{
+		Clusters: []clientcmdv1.NamedCluster{{
+			Name: "local",
+			Cluster: clientcmdv1.Cluster{
+				Server:                   apiServerURL,
+				CertificateAuthorityData: caCert,
+			}},
+		},
+		AuthInfos: []clientcmdv1.NamedAuthInfo{{
+			Name: "kubelet",
+			AuthInfo: clientcmdv1.AuthInfo{
+				Token: string(token),
+			},
+		}},
+		Contexts: []clientcmdv1.NamedContext{{
+			Name: "kubelet",
+			Context: clientcmdv1.Context{
+				Cluster:  "local",
+				AuthInfo: "kubelet",
+			},
+		}},
+		CurrentContext: "kubelet",
+	}
+	return &kubeconfig, nil
+}
+
+// createKubeletConf creates config file for kubelet, with Windows specific configuration.
+func (nc *nodeConfig) createKubeletConf(clusterDNS string) error {
+	kubeletConfig := generateKubeletConfiguration(clusterDNS)
+	kubeletConfigData, err := json.Marshal(kubeletConfig)
+	if err != nil {
+		return err
+	}
+	kubletConfigFileName := "kubelet.conf"
+	if err := nc.Windows.EnsureFileContent(kubeletConfigData, kubletConfigFileName, windows.K8sDir); err != nil {
+		return err
+	}
+	nc.log.V(1).Info("created", "file", windows.KubeletConfigPath)
+	return nil
+}
+
+// generateKubeletConfiguration returns the configuration spec for the kubelet Windows service
+func generateKubeletConfiguration(clusterDNS string) kubeletconfig.KubeletConfiguration {
+	falseBool := false
+	kubeAPIQPS := int32(50)
+	return kubeletconfig.KubeletConfiguration{
+		TypeMeta: meta.TypeMeta{
+			Kind:       "KubeletConfiguration",
+			APIVersion: "kubelet.config.k8s.io/v1beta1",
+		},
+		RotateCertificates: true,
+		ServerTLSBootstrap: true,
+		Authentication: kubeletconfig.KubeletAuthentication{
+			X509: kubeletconfig.KubeletX509Authentication{
+				ClientCAFile: windows.K8sDir + KubeletClientCAFilename,
+			},
+			Anonymous: kubeletconfig.KubeletAnonymousAuthentication{
+				Enabled: &falseBool,
+			},
+		},
+		ClusterDomain:         "cluster.local",
+		ClusterDNS:            []string{clusterDNS},
+		CgroupsPerQOS:         &falseBool,
+		RuntimeRequestTimeout: meta.Duration{Duration: 10 * time.Minute},
+		MaxPods:               250,
+		KubeAPIQPS:            &kubeAPIQPS,
```

Comment:
Theres a lot of magic numbers used in this config. If the answer for why these numbers are being used is, thats what is done for linux workers, thats fine, just comment that at the start of the function.

Reply:
Sure, I took a look at the `kubelet.conf` on a linux worker and confirmed this